addons/mail/models/mail_cc_mixin.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, tools
+
+
+class MailCCMixin(models.AbstractModel):
+    _name = 'mail.cc.mixin'
+    _inherit = 'mail.thread'
+    _description = 'Email CC management'
+
+    email_cc = fields.Char('Email cc', help='List of cc from incoming emails.')
+
+    def _email_dict(self, cc_string):
+        '''return a dict of sanitize_email:raw_email from a string of cc'''
+        if not cc_string:
+            return {}
+        return {tools.email_split(cc)[0]: cc for cc in [cc.strip() for cc in cc_string.split(',')] if tools.email_split(cc)}
+
+    @api.model
+    def message_new(self, msg_dict, custom_values=None):
+        if custom_values is None:
+            custom_values = {}
+        cc_values = {
+            'email_cc': ", ".join(self._email_dict(msg_dict.get('cc')).values()),
+        }
+        cc_values.update(custom_values)
```

Comment:
Just personal preference, but it's easier to understand code with `custom_values.update(cc_values)` instead

Reply:
Maybe you do that because you do not want to change in place the passed dict `custom_values`? Would name variable `cc_values` differently then, something like `new_custom_values`?